src/js/components/search/filters/location/LocationPicker.jsx

```diff
@@ -51,39 +53,59 @@ export default class LocationPicker extends React.Component {
         const countryChanged = (prevProps.country.code !== this.props.country.code);
-        const isCityInState = ( // if selected city is inside the selected state, don't clear the selected city!
+        const isCityInState = ( // selected city is w/in the selected state
+            this.props.country.code === 'USA' &&
             this.props.state.code === this.props.city.code &&
             this.props.state.code && this.props.city.code
         );
 
+        const cityDeselected = (prevProps.city.name && !this.props.city.name);
+
         if (countryChanged && this.props.country.code === "USA") {
             // user has selected USA, load the state list
             this.props.loadStates();
-            this.props.clearCitiesAndSelectedCity();
+            if (manuallyPopulatedCountryChanged) {
+                this.props.clearCitiesAndSelectedCity();
+            }
         }
         else if (countryChanged && prevProps.country.code === 'USA') {
             // the user previously selected USA, need to clear these out
             this.props.clearStates();
-            this.props.clearCitiesAndSelectedCity();
+            if (manuallyPopulatedCountryChanged) {
+                this.props.clearCitiesAndSelectedCity();
+            }
         }
-        else if (countryChanged) {
+        else if (manuallyPopulatedCountryChanged) {
             //  since USA isn't selected and wasn't previously selected, only clear cities
             this.props.clearCitiesAndSelectedCity();
         }
         if (stateChanged && this.props.state.code) {
-            // state code changed, load the counties & districts
+            // new state selected , load the corresponding counties & districts
             this.props.loadCounties(this.props.state.code.toLowerCase());
             this.props.loadDistricts(this.props.state.code.toLowerCase());
-            if (!isCityInState) {
+            if (!isCityInState) { // only clear the city if the new state does not contain selected city
                 this.props.clearCitiesAndSelectedCity();
             }
         }
         else if (stateChanged && !this.props.state.code) {
+            // manually selected state was removed, clear counties, districts & cities
             this.props.clearCounties();
             this.props.clearDistricts();
-            this.props.clearCitiesAndSelectedCity();
+            if (manuallyPopulatedStateChanged) {
+                this.props.clearCitiesAndSelectedCity();
+            }
+        }
+
+        if (cityDeselected && this.props.state.autoPopulated) {
+            // city was deselected which auto populated state selection, so clear the state selection
+            this.props.selectEntity('state', defaultLocationValues.state);
+        }
+        else if (cityDeselected && this.props.country.autoPopulated) {
+            this.props.selectEntity('country', { code: "FOREIGN", name: "ALL FOREIGN COUNTRIES", autoPopulated: true });
         }
     }
```

Comment:
This is for 2787 and 2904; handling deselect after auto-populate on city selection.